common4j/src/main/com/microsoft/identity/common/java/opentelemetry/CertBasedAuthTelemetryHelper.java

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.opentelemetry;
+
+import javax.annotation.Nullable;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Scope;
+import lombok.NonNull;
+
+/**
+ * Assists classes associated with Certificate Based Authentication (CBA) with
+ *  telemetry-related tasks.
+ */
+public class CertBasedAuthTelemetryHelper {
+
+    private static Scope mScope;
```

Comment:
I'm not sure why this needs to be static. Why can't it be a member variable? Perhaps you intended to make it member given that you've prefixed it with `m`?

Reply:
Making `mScope` (should actually be renamed to `sScope` - my bad) a member variable would mean that I would need to create an instance of this class and pass it through as a new parameter to classes such as the challenge handlers, the cert based auth managers, and the smartcard sessions as well. Alternatively, I could make `sScope` a member variable of the challenge handlers, but that would also require passing it in as a parameter to the manager and and smartcard sessions, and I might as well rewrite this PR to instead collect variable values until the very end (where the span can be created and ended in one shot).
This is why I think having a static scope variable would be more appropriate, but let me know if there's something I'm missing here.

Reply:
Statics in general are considered bad because they represent global state and as far as I can see your CBA spans are not something global but rather each interaction with CBA is one Event or One Span. You can read more why statics are bad over the internet, here's one example: https://stackoverflow.com/questions/7026507/why-are-static-variables-considered-evil#:~:text=Static%20variables%20are%20generally%20considered,assumptions%20of%20object%2Doriented%20programming.

Here are some questions for you:

- How does a static scope variable represent the scope for one given CBA session? 
- What happens when a second session launches? 
- Does the scope get overridden? 
- How do we know which scope corresponds to which CBA session?